src/app/statsGraphs/PieChartGraph.tsx

```diff
@@ -0,0 +1,142 @@
+"use client";
+import React from "react";
+import { PieChart, Pie, Cell, Legend, ResponsiveContainer } from "recharts";
+import { fallbackColor, rankedColors } from "./utils";
+
+type PageSegment = {
+  name: string;
+  value: number;
+  color: string;
+};
+
+interface PieChartGraphProps {
+  data: Record<string, number>;
+}
+
+const PieChartGraph = ({ data }: PieChartGraphProps) => {
+  const sortedEntries = Object.entries(data).sort((a, b) => b[1] - a[1]);
+
+  const categoryData: PageSegment[] = sortedEntries.map(
+    ([name, value], index) => ({
+      name,
+      value,
+      color: rankedColors[index] || fallbackColor,
+    })
+  );
+
+  const renderCustomLabel = (props: any) => {
```

Comment:
Maybe there's a better way to type the props here? Libraries usually export types for such handlers, something like `BaseAxisProps['label']` here. This ensures that it will break when you e.g. update the underlying chart library if it changes the way such properties are handled.

Reply:
Also, I believe all the properties used here come from the handler - you can extract this function outside of the component to improve readability.